### PR TITLE
Initial implementation of Fastpass for Bazel

### PR DESCRIPTION
**Pull request by Duhemm**
This commit implements a port of Fastpass that creates Bloop projects
based on a Bazel build.

### CODE REVIEW

**Review by wiwa — CHANGES_REQUESTED** [review]
I started with a couple nits, but is it possible to split this PR into:  

1. Changes to the build / `.yml` files  
2. Addition of Bazel protobuf defs
3. Refactorings (e.g. seems like you refactored `BloopPants`) 
4. Additions to the logic for Bazel

Not sure how much effort it would take though, so not exactly a demand.

File: fastpass/src/main/scala/scala/meta/internal/fastpass/IO.scala [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  def copy(in: InputStream, out: OutputStream, bufSize: Int = 16384): Unit = {
+    var read: Int = -1
+    val buffer = new Array[Byte](bufSize)
+    while ({ read = in.read(buffer); read } != -1) {
+      out.write(buffer, 0, read)
+    }
+  }
+
+  def toFileName(
```
[the remark this answers is not in the document]
Comment (Duhemm) [reply fastpass/src/main/scala/scala/meta/internal/fastpass/IO.scala]: Turns out it wasn't used anymore 🎉

File: fastpass/src/main/scala/scala/meta/internal/fastpass/IO.scala [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      name: String,
+      suffix: String = "",
+      maxLength: Int = 255
+  ): String = {
+    val availableLength = maxLength - suffix.length
+    name.replace("/", ".").replace(":", ".").take(availableLength) + suffix
+  }
+
+  def makeReadableFilename(target: String): String = {
```
[the remark this answers is not in the document]
Comment (Duhemm) [reply fastpass/src/main/scala/scala/meta/internal/fastpass/IO.scala]: It's just moving code from the previous BloopPants. It'd prefer to keep it like it is at the moment if you don't mind.

File: fastpass/src/main/scala/scala/meta/internal/fastpass/IO.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (Duhemm) [reply fastpass/src/main/scala/scala/meta/internal/fastpass/IO.scala]: It was used actually. Weird. I changed it!

File: fastpass/src/main/scala/scala/meta/internal/fastpass/IO.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (wiwa) [reply fastpass/src/main/scala/scala/meta/internal/fastpass/IO.scala]: Yep that's fine.

File: fastpass/src/main/scala/scala/meta/internal/fastpass/bazelbuild/ActionGraph.scala [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      .flatMap(x => idToArtifact.get(x))
+  }
+
+  def transitiveInputsOf(
+      label: String,
+      mnemonics: Set[String]
+  ): List[Artifact] = {
+    for {
+      action <- labelToActions.getOrElse(label, Nil)
+      if mnemonics.contains(action.getMnemonic())
```
Comment (wiwa) [inline fastpass/src/main/scala/scala/meta/internal/fastpass/bazelbuild/ActionGraph.scala:29]: `mnemonics.get(action).map(transitiveInputsOf).getOrElse(Nil)`? I'm surprised that this actually compiles... does it desugar to just a `filter`? If so, it's fine if you leave it. On the other hand, the `for` comp can be in `Option` and the `getOrElse` could be in the end.

**Review by wiwa — APPROVED** [review]
LGTM as far as I can tell at least. Thanks!

**Pull request merged**